aws/resource_aws_appconfig_environment.go

```diff
@@ -0,0 +1,240 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigEnvironmentCreate,
+		Read:   resourceAwsAppconfigEnvironmentRead,
+		Update: resourceAwsAppconfigEnvironmentUpdate,
+		Delete: resourceAwsAppconfigEnvironmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"monitor": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alarm_arn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"alarm_role_arn": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppconfigEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	applicationID := aws.String(d.Get("application_id").(string))
+	environmentName := aws.String(d.Get("name").(string))
+	environmentDescription := aws.String(d.Get("description").(string))
+
+	var monitorList []*appconfig.Monitor
+	if monitors := d.Get("monitor").(*schema.Set).List(); len(monitors) > 0 {
```

Comment:
I think this takes an array of ARNs it looks like and not a block

Reply:
Yeah, I had to change this. Monitors are actually objects, not just arns, so creating an array of arns doesn't work. Instead I changed the schema a bit. Now monitors are defined like 

```
monitor {
alarm_arn = "arn"
alarm_role_arn = "arn2"
}
monitor {
alarm_arn = "arn"
alarm_role_arn = "arn2"
}
```
I also changed to the singular since each block is 1 monitor. Just like validators, I had to do the conversion from a map to an actual Monitor object.